common/src/test/java/com/microsoft/identity/common/internal/ui/webview/certbasedauth/DialogHolderTest.java

```diff
@@ -0,0 +1,201 @@
+                    @Override
+                    public void onCancel() {
+                        // If we get here, the test has passed.
+                    }
+                }
+        );
+    }
+
+    @Test
+    @Config(shadows={ShadowSmartcardCertPickerDialog.class})
+    public void testExceptionThrownFromCertPickerDialog() {
+        final List<ICertDetails> certList = new ArrayList<>();
+        mDialogHolder.showCertPickerDialog(
+                certList,
+                new SmartcardCertPickerDialog.PositiveButtonListener() {
+                    @Override
+                    public void onClick(@NonNull ICertDetails selectedCertDetails) {
+                        // This shouldn't be used.
+                        fail("Should not be here");
+                    }
+                },
+                new ICancelCbaCallback() {
+                    @Override
+                    public void onCancel() {
+                        // If we get here, the test has passed.
+                    }
+                }
+        );
+    }
+
+    @Test
+    @Config(shadows={ShadowSmartcardPinDialog.class})
+    public void testExceptionThrownFromPinDialog() {
+        mDialogHolder.showPinDialog(
+                new SmartcardPinDialog.PositiveButtonListener() {
+                    @Override
+                    public void onClick(final char[] pin) {
+                        // This shouldn't be used.
+                        fail("Should not be here");
+                    }
+                },
+                new ICancelCbaCallback() {
+                    @Override
+                    public void onCancel() {
+                        // If we get here, the test has passed.
+                    }
+                }
+        );
+    }
+
+    @Test
+    @Config(shadows={ShadowSmartcardErrorDialog.class})
+    public void testExceptionThrownFromErrorDialog() {
+        // We can't easily verify the callback is called since it's an anonymous inner class
+        // in the implementation, but we can verify no exceptions are thrown
+        mDialogHolder.showErrorDialog(
+                1,
+                1
+        );
+    }
+
+    @Test
+    @Config(shadows={ShadowSmartcardErrorDialog.class})
+    public void testExceptionThrownFromErrorDialogWithCustomButton() {
+        // We can't easily verify the callback is called since it's an anonymous inner class
+        // in the implementation, but we can verify no exceptions are thrown
+        mDialogHolder.showErrorDialog(
+                1,
+                1,
+                1
+        );
+    }
+
+    @Test
+    @Config(shadows={ShadowSmartcardPromptDialog.class})
+    public void testExceptionThrownFromSmartcardPromptDialog() {
+        mDialogHolder.showSmartcardPromptDialog(
+                new ICancelCbaCallback() {
+                    @Override
+                    public void onCancel() {
+                        // If we get here, the test has passed.
```

Comment:
Not sure if necessary, but we can consider having a latch after the .showSmartcardPromptDialog, that is unlocked inside `onCancel()`. That way, test will not pass unless `onCancel()` is called. I think you can set a timeout too so test doesn't run forever if it doesn't hit the callback

Reply:
Great suggestion; implemented in the latest commit.